contracts/Strategy.sol

```diff
@@ -557,6 +559,31 @@ contract Strategy is BaseStrategy {
     }
 
     // ********************** CALCS **********************
+    function ethToWant(uint256 _amtInWei)
+        public
+        view
+        override
+        returns (uint256)
+    {
+        if (_amtInWei == 0) {
+            return 0;
+        }
+        address[] memory path = new address[](2);
+        path[0] = address(want);
+        path[1] = address(WETH);
+
+        uint256[] memory amounts = router.getAmountsOut(_amtInWei, path);
+        return amounts[amounts.length - 1];
+    }
+
+    function liquidateAllPositions()
+        internal
+        override
+        returns (uint256 _amountFreed)
+    {
+        // NOTE: we try to unlock all of the collateral in the strategy (which should be == totalDebt)
+        (_amountFreed, ) = liquidatePosition(balanceOfWant());
```

Comment:
Should this not be instead:

`liquidatePosition(estimatedTotalAssets())`;  ??

if setEmergencyExit is called to go thru this path it will miss the debt part of the assets and report that as a loss unless im missing smth?

Reply:
Maybe this should be 

```
uint256 debtOutstanding = vault.debtOutstanding();
liquidatePosition(debtOutstanding);
```

Like in the normal path?

Reply:
it is a good comment.

staked SNX is kept in the users' wallet (strategy's wallet in this case), but locked against transfers. so we are basically asking to unlock full SNX balance, which is the maximum amount we will be able to retrieve. for that, the strategy will automatically repay debt (issued sUSD)